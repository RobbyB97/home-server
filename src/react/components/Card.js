import React from 'react'
import {Link} from 'react-router-dom'

const Card = ({image, title, subtext, link}) => (
    <div>
        <div className="card__wrap">
            <a className="card__link" href={link}>
                <div className="card">
                    <img src={image}/>
                    <div className="card__text">
                        <h2>{title}</h2>
                        <p>{subtext}</p>
                    </div>
                </div>
            </a>
        </div>
    </div>
)

export default Card