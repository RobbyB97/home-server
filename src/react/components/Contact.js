import React from 'react';


export class Contact extends React.Component {
    render() {
        return (
            <section id="Contact" className="contact">
                <div className="contact__shape one"></div>
                <div className="contact__shape two"></div>
                Contact component
            </section>
        );
    }
}


export default Contact;