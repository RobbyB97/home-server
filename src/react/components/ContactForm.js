/* Libraries */
import React from 'react'


export class ContactForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            nameValue: ''
        };

        this.handleNameChange = this.handleNameChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleNameChange(event) {
        this.setState({
            ...this.state,
            nameValue: event.target.value
        });
    }

    handleSubmit(event) {
        console.log('hi')
    }

    render() {
        return (
            <div className="ContactForm">
                <br /><br /><br /><br /><br />
                <form id="contactForm">
                    <p>
                        <label>Name</label>
                        <input type="text" 
                            value={this.state.nameValue} 
                            onChange={this.handleNameChange} 
                            required 
                        />
                    </p>
                </form>
            </div>
        );
    }
}


export default ContactForm