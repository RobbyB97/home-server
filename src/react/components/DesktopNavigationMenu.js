import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {setTitle} from '../actions/title'


const DesktopNavigationMenu = ({dispatch}) => (
    <div id="desktop-navigation-menu">
        <ul>
            <li>
                <NavLink to="/react/" onClick={() => {
                    dispatch(setTitle(''))
                }}>Home</NavLink>
            </li>
            <li>
                <NavLink to="/react/style_guide" onClick={() => {
                    dispatch(setTitle('Style Guide'))
                }}>Style Guide</NavLink>
            </li>
        </ul>
    </div>
)

export default connect()(DesktopNavigationMenu)