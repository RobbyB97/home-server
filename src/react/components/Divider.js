import React from 'react';


export class Divider extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div className={`divider ${this.props.section}`}>
                <div className="divider__shape one"></div>
                <div className="divider__shape two"></div>
            </div>
        );
    }
}


export default Divider;