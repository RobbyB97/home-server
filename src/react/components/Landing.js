import React from 'react';
import Particles from 'react-particles-js';


export class Landing extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            // Used for shape animation on bottom of landing section
            shapeOne: false,
            shapeTwo:  false,
            shapeThree: false,
            shapeFour: false,
            shapeFive: false
        }
    }

    componentDidMount() {
        this.landingAnimation()
    }

    landingAnimation = () => {
        setTimeout(() => {
            this.setState({
                shapeOne: true
            });    
        }, 500);
        setTimeout(() => {
            this.setState({
                shapeTwo: true
            });
        }, 650);
        setTimeout(() => {
            this.setState({
                shapeThree: true
            });
        }, 800);
        setTimeout(() => {
            this.setState({
                shapeFour: true
            });
        }, 950);
        setTimeout(() => {
            this.setState({
                shapeFive: true
            });
        }, 1100);
    }

    render() {
        return (
            <div className="landing">
                <div className="landing__background"></div>

                <div className="landing__tagline">
                    <h2>
                        Robby Bergers
                    </h2>
                    <p className="one">
                        Designer
                    </p>
                    <p className="two">
                        Developer
                    </p>
                </div>

                <Particles 
                    width="100vw"
                    height="100vh"
                    className="landing__particles"
                    params={{
                        particles: {
                            number: {
                                value: 150,
                                density: {
                                    enable: true,
                                    value_area: 700
                                }
                            },
                            line_linked: {
                                enable: true,
                                distance: 60,
                                color: '#808080',
                                opacity: 0.5,
                                width: 1
                            },
                            color: {
                                value: '#000'
                            }
                        }
                    }}
                />

                <div className={this.state.shapeOne ?
                    "landing__shape one":
                    "landing__shape"}></div>
                <div className={this.state.shapeTwo ?
                    "landing__shape two":
                    "landing__shapte"}></div>
                <div className={this.state.shapeThree ?
                    "landing__shape three":
                    "landing__shape"}></div>
                <div className={this.state.shapeFour ?
                    "landing__shape four":
                    "landing__shape"}></div>
                <div className={this.state.shapeFive ?
                    "landing__shape five":
                    "landing__shape"}></div>
                <div className="landing__bottom">
                    
                    
                </div>
            </div>
        );
    }
}


export default Landing;