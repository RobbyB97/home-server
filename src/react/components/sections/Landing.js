import React from 'react';
import Particles from 'react-particles-js';


export class Landing extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            emailModal: false
        };
    }

    componentDidMount() {
        setTimeout(() => {
            const thing = document.querySelector('p.left');
            
        }, 1000)
    }

    /* Controls email modal */
    toggleEmail() {
        this.setState({
            emailModal: !this.state.emailModal
        });
    }

    render() {
        return (
            <section id="Landing" className="landing">
                {/* Background color blocks */}
                <div className="landing__background">
                    <div className="one">
                        <div className="texture">
                            <div></div>
                        </div>
                    </div>

                    <div className="two">
                        <div className="texture">
                            <div></div>
                        </div>
                    </div>

                    <div className="three"></div>
                    
                    <div className="four">
                        <Particles 
                            width="100vw"
                            height="100vh"
                            className="landing__particles"
                            params={{
                                particles: {
                                    number: {
                                        value: 110,
                                        density: {
                                            enable: true,
                                            value_area: 700
                                        }
                                    },
                                    line_linked: {
                                        enable: true,
                                        distance: 200,
                                        color: '#f8f8f8',
                                        opacity: 0.5,
                                        width: 1
                                    },
                                    color: {
                                        value: '#f8f8f8'
                                    }
                                }
                            }}
                        />
                    </div>
                    
                    <div className="five"></div>
                </div>
                    
                {/* Tagline */}
                <header className="landing__tagline">
                    <h2 className="left">
                        Robby
                    </h2>

                    <h2 className="right">
                        Bergers
                    </h2>

                    <p className="left">
                        Designer
                    </p>

                    <p className="right">
                        Developer
                    </p>
                </header>

                {/* Link icons */}
                <nav className="landing__contact">
                    <a className="landing__github" href="https://www.github.com/robbyb97">
                        <div></div>
                    </a>
                    
                    <a className="landing__linkedin" href="https://www.linkedin.com/in/robbyb97/">
                        <div></div>
                    </a>

                    <a className="landing__upwork" href="https://www.upwork.com/o/profiles/users/~01f696e65011a54866/">
                        <div></div>
                    </a>

                    <a className="landing__twitter" href="https://twitter.com/RobbyBergers">
                        <div></div>
                    </a>
                </nav>

                {/* Menu buttons */}
                <nav className="landing__menu">
                    <a className="landing__view" href="#About">
                        See my work
                    </a>

                    <a className="landing__resume" target="_blank" href="https://bergers.dev/resume.pdf">
                        View my resume
                    </a>
                    
                    <a className="landing__email"
                        onClick={() => {
                            this.toggleEmail();
                        }}
                    >
                        Get in touch
                    </a>
                </nav>

                {/* Email modal */}
                <div 
                    className="landing__emailModal"
                    data-active={this.state.emailModal}
                >
                    <p>Email me @ <a href="mailto:bergersr97@gmail.com">bergersr97@gmail.com</a></p>
                </div>
            </section>
        );
    }
}


export default Landing;