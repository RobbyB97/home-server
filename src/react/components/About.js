import React from 'react';

import Divider from './Divider'


export class About extends React.Component {
    render() {
        return (
            <section id="About" className="about">

                <div className="about__content one">
                    <div className="about__portrait">
                        <img src="/dist/images/aboutImage.png" />
                        <div className="about__text one">
                            <p>
                                <span className="one">Hi, my name's Robby. <span className="two">I'm a front-end developer and digital artist</span>.</span>
                            </p>
                        </div>
                    </div>
                </div>

                <Divider section="about two" />

                <div className="about__content two">
                    <div className="about__text two">
                        <p>
                            I want to help you succeed by building a beautiful, fast and functional website for your business.
                        </p>
                    </div>
                </div>
            </section>
        );
    }
}


export default About;