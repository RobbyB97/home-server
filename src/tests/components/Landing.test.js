import React from 'react'
import {shallow} from 'enzyme'

import {Landing} from '../../react/components/Landing'


test('Render Landing', () => {
    const wrapper = shallow(<Landing />)
    expect(wrapper).toMatchSnapshot()
})