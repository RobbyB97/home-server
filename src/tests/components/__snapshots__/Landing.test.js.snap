// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Landing 1`] = `
<div
  className="landing"
>
  <div
    className="landing__background"
  />
  <div
    className="landing__tagline"
  >
    <h2>
      Robby Bergers
    </h2>
    <p
      className="one"
    >
      Designer
    </p>
    <p
      className="two"
    >
      Developer
    </p>
  </div>
  <t
    className="landing__particles"
    height="100vh"
    params={
      Object {
        "particles": Object {
          "color": Object {
            "value": "#000",
          },
          "line_linked": Object {
            "color": "#808080",
            "distance": 60,
            "enable": true,
            "opacity": 0.5,
            "width": 1,
          },
          "number": Object {
            "density": Object {
              "enable": true,
              "value_area": 700,
            },
            "value": 150,
          },
        },
      }
    }
    style={Object {}}
    width="100vw"
  />
  <div
    className="landing__shape"
  />
  <div
    className="landing__shapte"
  />
  <div
    className="landing__shape"
  />
  <div
    className="landing__shape"
  />
  <div
    className="landing__shape"
  />
  <div
    className="landing__bottom"
  />
</div>
`;
